src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```
@@ -171,20 +161,18 @@ public void write(byte[] b) throws IOException {
       totalWritten = totalWritten + b.length;
       LOG.trace("Write {} len {}. Total {}", mUrl.toString(), b.length, totalWritten);
     }
-    checkThreadState();
     mOutputStream.write(b);
   }
 
   @Override
   public void close() throws IOException {
-    checkThreadState();
     LOG.debug("HTTP PUT close {}", mUrl.toString());
     flush();
     mOutputStream.close();
     try {
-      writeThread.join();
-    } catch (InterruptedException ie) {
-      LOG.error(ie.getMessage());
+      futureTask.get();
+    } catch (Exception e) {
+      throw new IOException("Unable to complete write. Reason : " + e.getCause());
```

Don't swallow the exception, but add it as the second argument.